airflow/providers/google/suite/operators/drive.py

```diff
@@ -0,0 +1,118 @@
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""This file contains Google Drive operators"""
+
+import os
+from pathlib import Path
+from typing import TYPE_CHECKING, Optional, Sequence, Union
+
+from airflow.models import BaseOperator
+from airflow.providers.google.suite.hooks.drive import GoogleDriveHook
+
+if TYPE_CHECKING:
+    from airflow.utils.context import Context
+
+
+class GoogleDriveUploadOperator(BaseOperator):
+    """
+    Upload a list of files to a Google Drive folder.
+    This operator uploads a list of local files to a Google Drive folder.
+    The local files can be deleted after upload (optional)
+
+    :param local_paths: Python list of local file paths
+    :param drive_folder: path of the Drive folder
+    :param gcp_conn_id: Airflow Connection ID for GCP
+    :param delete: should the local files be deleted after upload?
+    :param chunk_size: File will be uploaded in chunks of this many bytes. Only
+        used if resumable=True. Pass in a value of -1 if the file is to be
+        uploaded as a single chunk. Note that Google App Engine has a 5MB limit
+        on request size, so you should never set your chunk size larger than 5MB,
+        or to -1.
+    :param resumable: True if this is a resumable upload. False means upload
+        in a single request.
+    :param delegate_to: The account to impersonate using domain-wide delegation of authority,
+        if any. For this to work, the service account making the request must have
+        domain-wide delegation enabled.
+    :param impersonation_chain: Optional service account to impersonate using short-term
+        credentials, or chained list of accounts required to get the access_token
+        of the last account in the list, which will be impersonated in the request.
+        If set as a string, the account must grant the originating account
+        the Service Account Token Creator IAM role.
+        If set as a sequence, the identities from the list must grant
+        Service Account Token Creator IAM role to the directly preceding identity, with first
+        account from the list granting this role to the originating account
+    :return: Remote file ids after upload
+    :rtype: Sequence[str]
+    """
+
+    template_fields = (
+        'local_paths',
+        'drive_folder',
+    )
+
+    def __init__(
+        self,
+        local_paths: Sequence[Path],
```

Comment:
I would also allow `Sequence[str]` to suport templating this field. I don't think jinja will resolve paths. 
```suggestion
        local_paths: Union[Sequence[Path], Sequence[str]],
```
And in constructor we need to map in case of strings

Reply:
the same applies to drive_folder as well.